doc/adr/0007-queryable-properties.md

```diff
@@ -0,0 +1,286 @@
+# 7. Queryable properties
```

Comment:
> General comment: I think this is a very clever proposal, but there are some aspects of it that I still do not fully understand. The main "fear" I feel with this proposal is that the trace-server will have to embed too much logic that is specific to only one specific client, and that results useless/redundant/confusing to many other clients. But probably it is just because I do not fully understand the logic, so pick this statement with a grain of salt.
> 
I take the `clever proposal` :-)

I glad that you raise your concerns. We use ADR to be able to discuss the proposals to come up with reasonable design. 

I understand that it feels complicated. I was wondering myself if it's over-engineered and if it could be made simpler. 

> In general I feel that the client should have a minimal intelligence too, to avoid overloading the server. For example, if the client asks data from the end point that represents a data provider that generates the latency statistics tables, I would expect that the client knows what it is actually requesting (i.e. a table that contains "_the statistics of the latencies. These show the minimum, maximum, average, standard deviation, count and Total of the latencies when applicable_" as documented [here](https://archive.eclipse.org/tracecompass/doc/stable/org.eclipse.tracecompass.doc.user/Latency-Analyses.html)). I mean, as we expect that a user reads the documentation before clicking randomly on trace compass views to get what he/she wants...

> 
> Any thoughts? Or am I diverging too much from the fact that the front-end should be a pure "data visualization tool without domain specific knowledge" ?

Not sure how we could convey some domain specific content to the client that. We would have to define some "Concepts" which instructs the clients with additional behaviour:
- If a column is tagged (or has metadata) 'maximum', then client can fetch that time range.  
- If an element has a string that is a timestamp or time range, user should be able to select and navigate to
- If an element has a string that corresponds to a callsite, the user should be able to open that callsite

The latter 2 could use the tooltip endpoint, where we have key-value pairs to be displayed and where the key is the known metadata string.

> 
> If I'd like to give a counter-proposal, should I create a separate ADR, or should we try to keep the discussion on this one?

Maybe a dedicated proposal would be easier to follow. You can use markdown formatting, provide sequence diagrams etc. to illustrate your proposal. We can have a reference in this ADR.